### PR TITLE
feat: add duration metric to responses, leaderboard and participation summary

### PR DESCRIPTION
**Pull request by KaustubhKumar05**
image

image

image

### CODE REVIEW

File: packages/roomkit-react/src/Prebuilt/components/Polls/Voting/QuestionCard.jsx [hunk 1]
```diff
@@ -48,27 +41,13 @@ export const QuestionCard = ({
 
   const isLive = pollState === 'started';
   const canRespond = isLive && !localPeerResponse;
-
+  const startTime = Date.now();
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/components/Polls/Voting/QuestionCard.jsx:44]: no chance of re-render here? probably should be in a ref.

File: packages/roomkit-react/src/Button/Button.tsx [hunk 2]
```diff
@@ -143,8 +143,8 @@ const StyledButton = styled('button', {
         '$secondary_bright',
         '$secondary_dim',
         '$secondary_disabled',
-        '$on_surface_high',
-        '$on_surface_low',
+        '$on_secondary_high',
```
Comment (raviteja83) [inline packages/roomkit-react/src/Button/Button.tsx:146]: this fixes the start recording button as well?

Comment (KaustubhKumar05) [reply packages/roomkit-react/src/Button/Button.tsx:146]: The tokens are now correct but proper color selection is also needed on dashboard by the user

File: packages/roomkit-react/src/Prebuilt/components/Polls/Voting/QuestionCard.jsx (hunk 1 above)
Comment (KaustubhKumar05) [reply packages/roomkit-react/src/Prebuilt/components/Polls/Voting/QuestionCard.jsx]: Updated and tested

File: packages/roomkit-react/src/Button/Button.tsx (hunk 2 above)
Comment (raviteja83) [reply packages/roomkit-react/src/Button/Button.tsx:146]: the color selection should be fine. as we show the preview there, we can put it on the user to be more mindful

File: packages/roomkit-react/src/Prebuilt/components/Polls/Voting/TimedVoting.tsx [hunk 3]
```diff
@@ -9,29 +9,31 @@ export const TimedView = ({ poll }: { poll: HMSPoll }) => {
...
-        answer={question.answer}
-        setCurrentIndex={setCurrentIndex}
-        rolesThatCanViewResponses={poll.rolesThatCanViewResponses}
-      />
-    ) : (
-      <></>
-    ),
+  return (
+    <>
+      {poll.questions?.map(question =>
```
Comment (raviteja83) [inline packages/roomkit-react/src/Prebuilt/components/Polls/Voting/TimedVoting.tsx:17]: why not use activeQuestion itslef here?

Comment (KaustubhKumar05) [reply packages/roomkit-react/src/Prebuilt/components/Polls/Voting/TimedVoting.tsx:17]: Had tried that first, rendering just one QuestionCard element. The radio group selection does not reset on rerender and selection from previous question remains. Tried the HTML input as well, this seems to be the expected behaviour

**Review by raviteja83 — APPROVED** [review]

**Pull request merged**